src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterface.cs

```diff
@@ -1791,10 +1791,16 @@ private string ReadLineFromConsole(bool endOnTab, string initialContent, bool ca
                     }
 
                     // Modify string
-                    if (!insertMode) // then overwrite mode
+                    if (!insertMode && s != "") // then overwrite mode
                     {
                         s = s.Remove(index, 1);
                     }
+
+                    if (index > s.Length)
```

Comment:
I think it would be helpful to have a comment explain when `index > s.Length` would happen.